tests/bazel_tests.bzl

```diff
@@ -151,20 +186,24 @@ def _test_environment_impl(ctx):
     bazel = home + "/.bazel/{0}/bin/bazel".format(ctx.os.environ["BAZEL_VERSION"])
   if bazel == "" or not ctx.path(bazel).exists:
     bazel = ctx.which("bazel")
-  if ctx.name != ws:
-    fail("workspace did not match, expected:", ctx.name, "got:", ws)
-  ctx.file("WORKSPACE", """
-workspace(name = "%s")
-""" % ctx.name)
-  ctx.file("BUILD.bazel", """
-load("@io_bazel_rules_go//tests:bazel_tests.bzl", "execroot")
-execroot(
-    name = "execroot",
-    path = "{0}",
-    bazel = "{1}",
-    visibility = ["//visibility:public"],
-)
-""".format(execroot, bazel))
+
+  # Get a temporary directory to use as our scratch workspace
+  result = env_execute(ctx, ["mktemp", "-d"])
+  if result.return_code:
+        fail("failed to create temporary directory for bazel tests: {}".format(result.stderr))
+  scratch_dir = result.stdout.strip()
+
+  # Work out where we are running so we can find externals
+  exec_root, _, _ = str(ctx.path(".")).rpartition("/external/")
+
+  # build the basic environment
+  ctx.file("WORKSPACE", 'workspace(name = "{}")'.format(ctx.name))
```

Comment:
Do we actually need a WORKSPACE file in this repo?

Reply:
If we don't make one bazel will do it for us...